### ISSUE TITLE
DPOTrainer deepspeed.initialize cause ref_model not fixed

### ISSUE THREAD

**Issue by shuoYan97** [huggingface/trl#1652]
When using DPOTrainer with deepspeed zero1, I find that print(model.training) will give True which means the model can be updated but the ref_model should be fixed from the DPO paper. Does that make sense?

The related code can be found in <https://github.com/huggingface/trl/blob/v0.8.6/trl/trainer/dpo_trainer.py#L451>. Although model.eval() is used here, it cannot change the value of model.training and affect the logits of inference.

Thanks

**Comment by shuoYan97** [comment]
When using accelerator.prepare_model(), everything seems fine. I am not very familiar with deepspeed, is there a way to make this if-else structure has the same behaviour for two conditions?

**Comment by JeffWb** [comment]
I have the same question too. I try to train a model with 32b params, even using dpotrainer with ds-zero3，50 nodes with 8 40g-A100 are not enough. I doubt that the ref_model.eval() doesnt work.

**Comment by DZ9** [comment]
> https://github.com/huggingface/trl/blob/v0.8.6/trl/trainer/dpo_trainer.py#L451

same issue

**Comment by skylooop** [comment]
Hi! Have you managed to solve this problem? During generate_during_eval=True, ref_model outputs are almost the same as current policy model trained. So it seems like ref_model is being trained along.
